File: src/main/java/io/tatum/model/response/ledger/Blockage.java
package io.tatum.model.response.ledger;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * The type Blockage.
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@EqualsAndHashCode
@JsonIgnoreProperties(ignoreUnknown = true)
public class Blockage {

    /**
     * ID of the blockage.
     * @type {string}
     * @memberof Blockage
     */
    private String id;

    /**
     * ID of the account this blockage is for.
     * @type {string}
     * @memberof Blockage
     */
    private String accountId;

    /**
     * Amount blocked on account.
     * @type {string}
     * @memberof Blockage
     */
    private String amount;

    /**
     * Type of blockage.
     * @type {string}
     * @memberof Blockage
     */
    private String type;

    /**
     * Description of blockage.
     * @type {string}
     * @memberof Blockage
     */
    private String description;

}

File: src/main/kotlin/io/tatum/model/response/ledger/Blockage.kt
package io.tatum.model.response.ledger

/**
 * Blockage
 *
 * @property id ID of the blockage.
 * @property accountId ID of the account this blockage is for.
 * @property amount Amount blocked on account.
 * @property type Type of blockage.
 * @property description Description of blockage.
 */
class Blockage(
    val id: String,
    val accountId: String,
    val amount: String,
    val type: String,
    val description: String? = null,
)
